File: 06/main.pl
#!/usr/bin/env perl

use strict;
use warnings;
$/ = ""; # paragraph mode

my ($part1, $part2) = (0, 0);
while (my $group = <>) {
    chomp $group;
    my @people = split /\n/, $group;
    my %questions;
    for my $q (split '', $group) {
        next unless $q =~ /\w/;
        $part2++ if ++$questions{$q} == scalar(@people);
    }
    $part1 += scalar(keys %questions);
}
print "PART1: $part1\n";
print "PART2: $part2\n";

File: 06/main.py
import sys
import re
from collections import Counter

(part1, part2) = (0, 0)
for group in re.split("\n\n", sys.stdin.read().rstrip()):
    people = re.split("\n", group)
    questions = Counter()
    for q in group:
        if not re.match("\w", q): continue
        questions[q] += 1
        if questions[q] == len(people): part2 += 1
    part1 += len(questions)

print("PART 1:", part1)
print("PART 2:", part2)
